src/scitokens_internal.cpp

```diff
@@ -37,8 +38,93 @@ std::mutex key_refresh_mutex;
 
 namespace scitokens {
 
+// Define the static once_flag for Validator
+std::once_flag Validator::m_background_refresh_once;
+
 namespace internal {
 
+// BackgroundRefreshManager implementation
+void BackgroundRefreshManager::start() {
+    std::lock_guard<std::mutex> lock(m_mutex);
+    if (m_running) {
+        return; // Already running
+    }
+    m_shutdown = false;
+    m_running = true;
+    m_thread = std::make_unique<std::thread>(
+        &BackgroundRefreshManager::refresh_loop, this);
+}
+
+void BackgroundRefreshManager::stop() {
+    std::unique_ptr<std::thread> thread_to_join;
+
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        if (!m_running) {
+            return; // Not running
+        }
+
+        m_shutdown = true;
+        m_running = false;
+        thread_to_join = std::move(m_thread);
+    }
+
+    m_cv.notify_all();
+
+    if (thread_to_join && thread_to_join->joinable()) {
+        thread_to_join->join();
+    }
+}
+
+void BackgroundRefreshManager::refresh_loop() {
+    while (!m_shutdown) {
```

Comment:
Do not use full sequential consistency semantics that is the default for atomics.  Instead, use acquire/release semantics.